core/src/main/scala/kafka/utils/ShutdownableThread.scala

```diff
@@ -63,6 +65,8 @@ abstract class ShutdownableThread(val name: String, val isInterruptible: Boolean
         doWork()
       }
     } catch{
+      case fatalExitError: FatalExitError =>
+        fatalExitError.shutdownSystem()
```

Comment:
I also was thinking the same thing but then it occurs to me that shutdown hook can finish successfully regardless the state of the anonymous thread that has initiated it. The anonymous thread would still join the thread that is running the shutdown hook, which should properly terminate after the shutdown hook thread terminates. If this understanding is correct, then making the anonymous thread a daemon should not make much difference.